mountpoint-s3/src/inode.rs

```diff
@@ -937,12 +937,11 @@ impl WriteHandle {
         let inode = inner.get(ino)?;
         let mut state = inode.get_mut_inode_state()?;
         match state.write_status {
-            WriteStatus::LocalUnopened => {
+            WriteStatus::LocalUnopened | WriteStatus::Remote => {
```

Comment:
Think we need to do something here with files that are already open for reading. If an application has a file open `O_RDONLY`, and then opens it again `O_WRONLY`, seems like here the second open would succeed, which isn't what we want.

Reply:
I was planning to do that as a separate change but might be a good idea to include it in this PR.

Reply:
I'm OK with doing it in a separate PR if you think it'll make reviewing easier, but let's put a TODO or open an issue or something.